packages/inventory-compliance/src/SystemRulesTable/SystemRulesTable.js

```diff
@@ -276,11 +279,14 @@ class SystemRulesTable extends React.Component {
         let selectedToRemediate;
 
         if (this.selectAllCheckbox(key)) {
-            selectedToRemediate = this.getRules().map((rule) => rule.rowKey);
+            let ruleRowKeys = this.getRules().map((rule) => rule.rowKey);
+            selectedToRemediate = selected ?
+                [ ...new Set([ ...currentlySelectToRemediate, ...ruleRowKeys ]) ] :
+                currentlySelectToRemediate.filter((rowKey) => !ruleRowKeys.includes(rowKey));
         } else {
             selectedToRemediate = selected ?
                 [ ...currentlySelectToRemediate, rowData.rowKey ] :
-                currentlySelectToRemediate.filter((refId) => (refId !== rowData.rowKey));
+                currentlySelectToRemediate.filter((rowKey) => (rowKey !== rowData.rowKey));
```

Comment:
I know, unnecessary, but this variable name was very confusing to me when fixing this bug. These row keys are not quite the same as refId. Here's an example rowKey: `'xccdf_org.ssgproject.content_rule_grub2_password-xccdf_org.ssgproject.content_profile_rht-ccp'`